include/sensors/core/tasks/pressure_driver.hpp

```diff
@@ -251,155 +253,218 @@ class MMR920C04 {
         can_client.send_can_message(get_host_id(), message);
     }
 
-    auto sensor_callback() -> void {
-        writer.transact_isr(mmr920C04::ADDRESS,
-                            static_cast<uint8_t>(read_register),
-                            static_cast<std::size_t>(3), own_queue,
-                            static_cast<uint8_t>(read_register));
-        if (limited_poll && !stop_polling) {
-            readings_taken++;
-            if (readings_taken >= total_baseline_reads) {
-                stop_polling = true;
-            }
-        }
-        if (stop_polling) {
-            writer.write_isr(mmr920C04::ADDRESS,
-                             static_cast<uint8_t>(mmr920C04::Registers::RESET));
-        }
+    void reset_readings() {
+        writer.write(mmr920C04::ADDRESS,
+                     static_cast<uint8_t>(mmr920C04::Registers::RESET));
     }
 
-    auto set_sync_bind(SensorOutputBinding binding) -> void {
-        hardware.reset_sync();
-        set_sync(binding);
-        set_report(binding);
-        set_stop_polling(binding);
+    void stop_continuous_polling(uint8_t transaction_index, uint8_t reg_id) {
+        poller.continuous_single_register_poll(mmr920C04::ADDRESS, reg_id, 3,
+                                               STOP_DELAY, own_queue,
+                                               transaction_index);
     }
 
-    auto handle_baseline_poll(uint32_t message_index, float data) -> void {
-        running_total += data;
-        if (readings_taken == total_baseline_reads) {
-            offset_average = running_total / total_baseline_reads;
-            send_baseline_response(message_index);
-            set_number_of_reads(0);
+    auto handle_ongoing_response(i2c::messages::TransactionResponse &m)
+        -> void {
+        auto reg_id = utils::reg_from_id<mmr920C04::Registers>(m.id.token);
+        if (!bind_sync && !echoing) {
+            stop_continuous_polling(m.id.transaction_index,
+                                    static_cast<uint8_t>(reg_id));
+            reset_readings();
+        }
+
+        // Pressure is always a three-byte value
+        static_cast<void>(bit_utils::bytes_to_int(m.read_buffer.cbegin(),
+                                                  m.read_buffer.cend(),
+                                                  temporary_data_store));
+
+        int32_t signed_data_store =
+            static_cast<int32_t>(temporary_data_store >> 8);
+
+        auto pressure_read =
```

Comment:
these can't both be true, right? can we factor this out into a function for each?